src/garage/envs/point_env.py

```diff
@@ -97,24 +125,60 @@ def step(self, action):
         # done bonus
         if succ:
             reward += self._done_bonus
+        # Type conversion
+        if not isinstance(reward, float):
+            reward = float(reward)
 
         # sometimes we don't want to terminate
         done = succ and not self._never_done
 
         obs = np.concatenate([self._point, (dist, )])
 
-        return Step(obs, reward, done, task=self._task, success=succ)
-
-    def render(self, mode='human'):
-        """Draw the environment.
-
-        Not implemented.
+        last_obs = self._last_observation
+        self._last_observation = last_obs
+        self._step_cnt += 1
+
+        step_type = None
+        if done:
+            step_type = StepType.TERMINAL
+        elif self._step_cnt >= self._spec.max_episode_length:
+            step_type = StepType.TIMEOUT
+        elif self._step_cnt == 1:
+            step_type = StepType.FIRST
+        else:
+            step_type = StepType.MID
+
+        return TimeStep(
+            env_spec=self.spec,
+            observation=last_obs,
+            action=action,
+            reward=reward,
+            next_observation=obs,
+            env_info={'task': self._task, 'success': succ},
+            agent_info={},  # TODO: can't be populated by env
+            step_type=step_type)
+
+    def render(self, mode):
+        """Renders the environment.
 
         Args:
-            mode (str): Ignored.
-
+            mode (str): the mode to render with. The string must be present in
+                `self.render_modes`.
         """
-        # pylint: disable=no-self-use
+        pass
+
+    def visualize(self):
+        """Creates a visualization of the environment."""
+        pass
```

Comment:
maybe just print the position of the point, goal, and L2 distance between them on each step?